oelint-adv: init at 3.21.4

christoph-heiss opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This adds [oelint-adv](https://github.com/priv-kweihmann/oelint-adv), a linter for BitBake files/recipes. Additionally adds [oelint-parser](https://github.com/priv-kweihmann/oelint-parser) as a Python 3 module, which is required by the former. 

~~The tests for both modules currently fail due to some weird issue I still have to figure out. I would submit an update later to fix it, or we block merging until this is resolved - I'm fine with either.~~

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

christoph-heiss committed:
python3Packages.oelint-parser: init at 2.10.5

Signed-off-by: Christoph Heiss <christoph@c8h4.io>

onny commented on nixos/doc/manual/release-notes/rl-2305.section.md:
```diff
@@ -68,6 +68,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - [imaginary](https://github.com/h2non/imaginary), a microservice for high-level image processing that Nextcloud can use to generate previews. Available as [services.imaginary](#opt-services.imaginary.enable).
 
+- [oelint-adv](https://github.com/priv-kweihmann/oelint-adv), a advanced linter for BitBake recipes based on oelint.
```
This part of the release notes is only for new services, not new packages. As far as I know we don't add new packages to the release notes

christoph-heiss replied:
Well, there are some other tools/packages listed, like [trurl](https://github.com/NixOS/nixpkgs/pull/224526), which was merged not that long ago.

But I'll remove it's preferred, I fine with either way.

onny reviewed (commented):
Thank you for this nice contribution, I added some small notes which would be nice to get fixed :)

christoph-heiss commented on pkgs/development/python-modules/oelint-parser/default.nix:
```diff
@@ -0,0 +1,38 @@
+  pythonImportsCheck = [ "oelint_parser" ];
+
+  # Currently fails due to various issues
+  doCheck = false;
```
I tried that, but did not realize I that `pytestCheckHook` must come _after_ all other `pytest`-dependencies in `checkInputs`, which otherwise result in some transient dependencies. Not exactly obvious 😓.

christoph-heiss commented:
Thanks for reviewing! Got around to have another look at it, solved the issue with the tests now!

christoph-heiss committed:
oelint-adv: init at 3.21.4

Two tests have to be disabled as they require network access, and the
coverage has to be lowered as required to pass.

Signed-off-by: Christoph Heiss <christoph@c8h4.io>

christoph-heiss commented:
Updated `oelint-parser` to 3.21.4 and removed the note from the release note (which was outdated anyway).
